### ISSUE TITLE
UI rendering freezes when writing in other languages.

### ISSUE THREAD

**Issue by Nashenas88** [pop-os/pop#877]
**Distribution (run `cat /etc/os-release`):**
```
NAME="Pop!_OS"
VERSION="19.10"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 19.10"
VERSION_ID="19.10"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=eoan
UBUNTU_CODENAME=eoan
LOGO=distributor-logo-pop-os
```
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
Not clear

**Issue/Bug Description:**
I've been trying to learn Russian, so I added the Russian (Macintosh) keyboard to my input sources.
![Screenshot from 2020-03-19 18-42-34](https://user-images.githubusercontent.com/1673130/77121419-87968b80-6a11-11ea-999b-8243e04fe37e.png)

When typing, after pressing enter (or a few seconds, not sure which), the UI will freeze.

I can see the mouse, it responds to hover events because it will change the cursor icon, e.g. if it's over a button or over text. But the UI itself will not redraw besides the mouse. No key combinations will make any noticeable change to the UI.

I can switch to a tty with Ctrl+Alt+F3, login and then sometimes running `sudo killall -3 gnome-shell` will get me back, but not always. In most cases I will see a "Oh no! Something has gone wrong.", which asks me to logout. Interestingly, if I hit the super key, it will bring up all the windows and the apps are still running in that case, I just can't access them.

**Steps to reproduce (if you know):**
It seems to be when pressing Enter while in that keyboard that the issue occurs, or at least after a few seconds of typing (not sure which yet)

**Expected behavior:**
No UI freeze.

**Other Notes:**
Device: 2019 System 76 Gazelle Pro 

I'm not sure what data I can collect that will be helpful.

**Comment by leviport** [comment]
Hmm, I haven't been able to reproduce this one. Are you using any particular programs when this happens? I've typed in Firefox, Gnome Terminal, and Firefox, hitting enter multiple times, and I saw no freezing.

**Comment by Nashenas88** [comment]
See the video for what I'm running into. What logs could I collect that can help with this?

**Comment by Nashenas88** [comment]
https://photos.app.goo.gl/hYLwMnQokEfMUQM49

**Comment by leviport** [comment]
It's not really a matter of logs unfortunately. If I can't reproduce it, I can't investigate it as a bug with Pop!_OS. It looks like you're typing in Firefox with that layout enabled in your video. Doing the same thing on the machines I can test does not freeze anything up. Does this happen with other applications? Text editors, terminals, etc.?

**Comment by Nashenas88** [comment]
So I was able to reproduce it with what I think is the issue. I also had installed `fcitx-mozc` for typing in Japanese, as well as `gnome-shell-extension-appindicator`. All those `killall -3 gnome-shell` must have caused the extensions to all turn off. When I uninstall `fcitx-mozc` (and `fcitx`), then I can no longer reproduce the issue. In order to easily switch between English and Japanese, I had the input in fcitx configured to the Mozc Japanese input (otherwise it was more complicated). Is it possible fcitx and the gnome input source selector were fighting over which keyboard should be active? Whenever it would freeze, if I was running `htop` in the 3rd terminal (`Ctrl+Alt+F3`) then for a brief moment I would notice that the Xorg process was using around 80% of the CPU. It would drop right away once it was no longer rendering.

To summarize:
`sudo apt install fcitx-mozc`
`sudo apt install gnome-shell-extension-appindicator`
`gnome-shell-extension-prefs`: Turn on at least `KStatusNotifierItem/AppIndicator Support` (from https://pop.system76.com/docs/status-icons)
Go to the fcitx input dropdown and select Configure:
![Screenshot from 2020-03-20 17-32-21](https://user-images.githubusercontent.com/1673130/77207948-10283100-6ad1-11ea-9e7b-9f052a953b88.png)
Then ensure that Mozc (Japanese) is one of the input methods configured.
![Screenshot from 2020-03-20 17-32-31](https://user-images.githubusercontent.com/1673130/77208261-d4da3200-6ad1-11ea-8a7c-b10b60eea6d2.png)

From there, select the normal input source switcher, and select Russian. On my system that generates an instant freeze.

**Comment by leviport** [comment]
> Is it possible fcitx and the gnome input source selector were fighting over which keyboard should be active?

That's certainly a possibility. Another one is the appindicator extension, as installed extensions sometimes cause unexpected behavior. Any reason you don't just use Gnome's built-in layout switcher? I use it for switching between Dvorak and Qwerty, and it's rock solid.

**Comment by Nashenas88** [comment]
For Japanese, it requires a more involved editor. I had never realized that there was (or maybe there wasn't) a mozc option that was configurable in the default language input. Now I'm able to access Japanese and Russian input through the default keyboard switcher. Sorry for all the trouble. Is this still a valid bug or should I just close this?

**Comment by leviport** [comment]
No worries, it was no trouble at all. I'm speculating that the bug is either with the appindicator extension or `fcitx-mozc`, so I'll close it for now. If it does seem to be Pop-related, we can reopen it in the future and investigate further. Thank you for the bug report though!
